### ISSUE TITLE
ERROR: "ReferenceError: SharedArrayBuffer is not defined"

### ISSUE THREAD

**Issue by mixiaoluo88** [kanaverse/kana#262]
Hi,

Thank you for your efforts in creating such great software. I managed to deploy this app locally using the command `docker run -it --rm -p 3000:3000 kana:latest`. The Dockerfile is as follows:
```
FROM node:latest

# Expose the app port
EXPOSE 3000

COPY . /kana
WORKDIR /kana

# Build a static version of the app for deployment
RUN npm i --include=dev --force 
RUN npm dedupe --force 

# build the app
RUN PUBLIC_URL="/kana" npm run build

# Start the app
CMD npm run start
```
However, I encountered an error message preventing me from performing analysis when I opened the page：
![image](https://github.com/user-attachments/assets/c808d8d8-f293-40da-a059-d1183f85e6b3)

I look forward to your response.
Hongyang Lvy

**Comment by LTLA** [comment]
**kana** relies on `SharedArrayBuffer` to perform multi-threaded analyses, but browsers disable this feature by default for security reasons. We get around this by using a service worker to provide cross-site isolation so that `SharedArrayBuffer` becomes available. 

My guess is that, in your case, the service worker configuration isn't quite working properly; probably something to do with a mismatch between the `PUBLIC_URL=/kana` and where the app is actually being served. @jkanche would know better, but from what I can see, if you want to mimic our GitHub Pages set-up, there shouldn't be a `/kana` subdirectory; you should be serving the built app from the root of your web server.

**Comment by jkanche** [comment]
I agree with @LTLA and also depends on where you are running docker (local laptop vs a remote server), the later might require SSL certificates installed on the server. The Dockerfile included in the repo is for the hosting the application, If you remove the public url, 

```docker
FROM node:latest

# Expose the app port
EXPOSE 3000

COPY . /kana
WORKDIR /kana

# Build a static version of the app for deployment
RUN npm i --include=dev --force 
RUN npm dedupe --force 

# build the app
RUN npm run build

# Start the app
CMD npm run start
```

**Comment by jkanche** [comment]
let us know if you run into any issues. Thank you!
